trajopt/include/trajopt/file_write_callback.hpp

```diff
@@ -0,0 +1,12 @@
+#pragma once
+#include <trajopt/problem_description.hpp>
+#include <trajopt_sco/optimizers.hpp>
+
+namespace trajopt
+{
+Optimizer::Callback WriteCallback(std::shared_ptr<std::ofstream> file,
+                                  TrajOptProbPtr prob,
+                                  Eigen::Isometry3d tcp = Eigen::Isometry3d::Identity(),
+                                  bool angle_axis = false,
+                                  bool degrees = false);
+}
```

Comment:
- tcp applies a static transform to the pose in line 43
- angle_axis is a switch to output angle axis rotations instead of quaternions (I imagine to help him evaluate the conical constraint)
- degrees switches angle axis to degrees instead of radians

I am fine with simplifying it if you want. Just let me know what you'd like to do.